src/models/pacman.c

```diff
@@ -81,18 +81,24 @@ void drawPacman(int count)
     }
 
     // Get new pacman position in grid
-    struct Position newPacmanGridPos = uiPosToGridPos(pacmanPosCopy);
+    struct Position newPacmanGridPos = getUiPosToGridPos(pacmanPosCopy);
 
     if (!arePositionEquals(pacmanGridPos, newPacmanGridPos))
     {
-
-        if (isObstacle(newPacmanGridPos))
+        // If pacman, just blit him at without updating his position
+        if (isPositionContainsMazeElement(newPacmanGridPos, WALL))
```

Comment:
Well said !!

With my previous comment, this line would become

`if (getMazeElementAt(newPacmanGridPos) == WALL)`

I think it's clearer this way

Reply:
[11aacd3](https://github.com/Livl-Corporation/livl-pacman/commit/11aacd36abf4138cba46cbf5e638747a83c9ee9c)